components/raftstore-v2/src/raft/peer.rs

```diff
@@ -450,6 +456,18 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
         self.peer_heartbeats.remove(&peer_id);
     }
 
+    /// Returns whether or not the peer sent heartbeat after the provided
+    /// deadline time.
+    #[inline]
+    pub fn peer_heartbeat_is_fresh(&self, peer_id: u64, deadline: &Instant) -> bool {
+        if let Some(last_heartbeat) = self.peer_heartbeats.get(&peer_id)
+            && *last_heartbeat >= *deadline
+        {
+            return true;
+        }
+        false
+    }
```

Comment:
You mean `matches!(self.peer_heartbeats.get(&peer_id), Some(last_heartbat) if *last_heartbeat >= *deadline)`?